extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -73,24 +74,26 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
+    SettableByteEntity<? extends ByteEntity> settableByteEntitySource = (SettableByteEntity<? extends ByteEntity>) source;
+    ByteEntity byteEntitySource = settableByteEntitySource.getEntity();
     // assume InputEntity is always defined in a single classloader (the kafka-indexing-service classloader)
     // so we only have to look it up once. To be completely correct we should cache the method based on classloader
     if (getHeaderMethod == null) {
       getHeaderMethod = KafkaUtils.lookupGetHeaderMethod(
-          source.getClass().getClassLoader(),
+          byteEntitySource.getClass().getClassLoader(),
           OpenCensusProtobufInputFormat.VERSION_HEADER_KEY
       );
     }
 
     try {
-      byte[] versionHeader = (byte[]) getHeaderMethod.invoke(source);
+      byte[] versionHeader = (byte[]) getHeaderMethod.invoke(byteEntitySource);
```

Comment:
`createReader` is now only called when input format is initialized or the schema changes, so it no longer gets invoked with every row.
This means we need to save the reference to the settable input entity, and only call `source.getEntity()` from within the `read()` method of the InputEntityReader itself, once the data is actually ready to be consumed.
Once suggestion would be to create an intermediate (e.g. anonymous class) InputEntityReader in this method, and have that delegate to the right protobuf reader at the time that read() is called.

Reply:
I've created an inner class. Let's discuss in: https://github.com/confluentinc/druid/pull/113